src/bp/ml/svm/addon/index.ts

```diff
@@ -79,23 +68,16 @@ class BaseSVM {
     return svm.predict_probability(inputs).probabilities
   }
 
-  predictProbabilities = (inputs: number[]): Q.Promise<number[]> => {
+  predictProbabilities = async (inputs: number[]): Promise<number[]> => {
     assert(!!this._clf, 'train classifier first')
     const dims = numeric.dim(inputs)
     assert((dims[0] || 0) > 0 && (dims[1] || 0) === 0, 'input must be a 1d array')
 
-    const deferred = Q.defer<number[]>()
-
     const svm = this._clf as NSVM
-
-    svm.predict_probability_async(inputs, res => deferred.resolve(res.probabilities))
-
-    return deferred.promise
+    return new Promise(resolve => svm.predict_probability_async(inputs, p => resolve(p.probabilities)))
```

Comment:
predict can't fail?  what about the call of the function in itself? (try/catch wrap)